app/models/sms/errors.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# models an error generated by the sms system
+module Sms
+  module Errors
+    # base error class
+    class Error < StandardError
+    end
+
+    # error indicating that the entire operation was not able to succeed
+    class FatalError < Error
+    end
+
+    # some of the operation succeeded but included some errors
+    class PartialError < Error
+    end
```

Comment:
Actually now that I'm looking at it the Sms errors are getting a bit messy here. We now have:

* Sms::Errors::Error (which gets raised and rescued as a base class in some places)
* Sms::Errors::PartialError
* Sms::Errors::FatalError
* Sms::DecodingError
* Sms::Parser::Error
* Sms::UnverifiedTokenError

Yikes!

Rather than an `Errors` namespace it seems like maybe we should have the errors live in the appropriate namespaces under `Sms`.

Seems to me we should maybe have:

* Sms::GenericError (base class, all others inherit from this, just rename existing Sms::Error)
* Sms::Adapters::PartialSendError (the new one, move to Adapters namespace)
* Sms::Adapters::FatalSendError (the new one, move to Adapters namespace)
* Sms::Decoder::DecodingError (move to namespace)
* Sms::Decoder::ParseError (change namespace and class name)
* Sms::UnverifiedTokenError (same namespace, just make sure it continues to inherit from GenericError)

To make this make sense I think the Parser namespace could be renamed to Decoder and Decoder can go in there (so it would be Sms::Decoder::Decoder). The Decoder and Parser are obviously highly related.

What do you think? I know this is a lot but if we don't keep this organized it's going to get ugly.